### ISSUE TITLE
Very large scroll search (i.e. reindex) can gradually slow down 

### ISSUE THREAD

**Issue by jakelandis** [elastic/elasticsearch#65780]
Since 7.7 (via this [PR](https://github.com/elastic/elasticsearch/pull/52822)) added better ability to cancel a search request. However, this resulted in [adding](https://github.com/elastic/elasticsearch/pull/52822/files#diff-9f5585442c620397550df0ceb9faf151e8042e3c7444254ebf562153073c1d3fR269) a method to cancel a task to a collection on the context searcher. That collection is [checked very frequently](https://github.com/elastic/elasticsearch/blob/v7.7.0/server/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java#L355) and the count of that collection can grow unbounded. The memory footprint is not an issue, rather the number of iterations for very long running scroll searches, such as used by re-index. In testing this started to show an issue around 50m documents and kept increasing the search latency as time went on.  

Below is a test run of 180m documents being re-index that show the increase in the search latency and decrease in the search rate. 

(7.9.1)
![image](https://user-images.githubusercontent.com/976291/100934058-cf801480-34b3-11eb-9056-93d93d47382a.png)

This issue is fixed as 7.10.0 due to https://github.com/elastic/elasticsearch/pull/61062 and https://github.com/elastic/elasticsearch/issues/46523 which will now re-create the searcher on each phase even for scroll requests. Which means that this collection will grow unbounded anymore.  The same test above was run on 7.10.0 and did not show any signs of performance degradation. 

For 7.7 -> 7.9.x there is an easy work around to for this issue:
```
PUT _cluster/settings
{
  "persistent": {
    "search.low_level_cancellation" : false
  }
}
```
Which will will prevent that collection from even being used. (also tested to fix the issue).

**Comment by jakelandis** [comment]
Since this issue is fixed in 7.10.0, I will close this issue as fixed by #61062 and this issue is for documentation purposes only.

**Comment by jakelandis** [comment]
related to slow reindex (but a totally different cause) https://github.com/elastic/elasticsearch/issues/65788
